Proteomics/ModificationWithLocation.cs

```diff
@@ -64,20 +64,11 @@ public override bool Equals(object o)
             return m == null ? false :
 
                base.Equals(m)
-               && (this.accession == null && m.accession == null
```

Comment:
I'd like modifications to be equal if they have the same properties, i.e. mass, chemical formula, etc. So even with different accessions, modifications might be equal. 
The reason to make this change is if we writing a modification from uniprot and then read it, the way it was those would not be equal, since the uniprot accession is not written. If we don't compare that, then writing and reading modification back would be equal to the original.